src/main/java/org/jenkinsci/plugins/workflow/cps/DSL.java

```diff
@@ -165,34 +160,24 @@ public Object invokeMethod(String name, Object args) {
                     throw new GroovyRuntimeException(x);
                 }
             }
-            Set<String> ambiguousFunctionNames = new HashSet<>(0);
+            unreportedAmbiguousFunctions = new HashSet<>(0);
             for (StepDescriptor d : StepDescriptor.all()) {
                 String functionName = d.getFunctionName();
                 if (functions.containsKey(functionName)) {
-                    ambiguousFunctionNames.add(functionName);
+                    unreportedAmbiguousFunctions.add(functionName);
                 }
-                // TODO: Switch we switch to putIfAbsent so that the descriptor with the highest ordinal value is preferred for ambiguous functions?
-                functions.put(functionName, d);
+                // The step descriptor with the highest value for Extension#ordinal() is used in the case of ambiguity.
+                functions.putIfAbsent(functionName, d);
                 stepClassNames.put(d.clazz.getName(), d);
             }
-            ambiguousFunctions = functions.keySet().stream()
-                    .filter(ambiguousFunctionNames::contains)
-                    .collect(Collectors.groupingBy(Function.identity(),
-                            Collectors.mapping(n -> functions.get(n).clazz.getName(), Collectors.toList())));
-            ambiguousFunctionsLogged = new HashSet<>(0);
         }
         final StepDescriptor sd = functions.getOrDefault(name, stepClassNames.get(name));
         if (sd != null) {
-            List<String> ambiguousClassNames = ambiguousFunctions.get(name);
-            if (ambiguousClassNames != null && ambiguousFunctionsLogged.add(name)) {
-                LOGGER.log(Level.WARNING, "Invoking ambiguous Pipeline Step ‘{0}’ ({1}) in {3}. " + 
-                    "The function name could refer to any of the following steps: {2}. " + 
-                    "To avoid ambiguity, you can invoke steps by class name. " +
-                    "For example: steps.''{1}''(... args ...)", new Object[] {
-                        sd.getFunctionName(), sd.clazz.getName(), ambiguousClassNames, handle
-                });
+            if (Util.isOverridden(DSL.class, getClass(), "invokeStep", StepDescriptor.class, Object.class) &&
+                    !Util.isOverridden(DSL.class, getClass(), "invokeStep", StepDescriptor.class, String.class, Object.class)) {
```

Comment:
ha, nice. Weird, probably in practice unneeded (I really don't think there's anything extending `DSL` out there anywhere), but why not? =)

Reply:
Aaaah, makes sense.